dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedIntervalIterableGenerator.java

```diff
@@ -0,0 +1,153 @@
+   * @throws MetadataDumperUsageException in case of arguments incompatibility or missing arguments
+   */
+  @Nonnull
+  public static ZonedIntervalIterable forConnectorArguments(@Nonnull ConnectorArguments arguments)
+      throws MetadataDumperUsageException {
+    return ZonedIntervalIterableGenerator.forConnectorArguments(
+        arguments, Duration.ofHours(1), TimeTruncator.createBasedOnChronoUnit(ChronoUnit.HOURS));
+  }
+
+  /**
+   * Builds a ZonedIntervalIterable from connector arguments with the specified interval. The
+   * intervals have inclusive starting datetime and exclusive ending datetime (i.e.: start <= t <
+   * end ).
+   *
+   * @param arguments connector arguments
+   * @param duration the length of the intervals
+   * @return a nonnull ZonedIntervalIterable
+   * @throws MetadataDumperUsageException in case of arguments incompatibility or missing arguments
+   * @throws IllegalArgumentException if then {@link Duration} is more than a day or does not divide
+   *     a day evenly
+   */
+  @Nonnull
+  public static ZonedIntervalIterable forConnectorArguments(
+      @Nonnull ConnectorArguments arguments,
+      @Nonnull Duration duration,
+      @Nonnull TimeTruncator truncator)
+      throws MetadataDumperUsageException {
+    Preconditions.checkArgument(
+        isValidDuration(duration),
+        "Invalid duration provided. Please make sure the duration is less than a day and"
+            + " divides a 24 hour period evenly.");
+
+    if (arguments.getQueryLogStart() != null || arguments.getQueryLogEnd() != null) {
+      if (arguments.getQueryLogDays() != null) {
+        throw new MetadataDumperUsageException(
+            "Incompatible options, either specify a number of log days to export or a start/end"
+                + " timestamp.");
+      }
+
+      if (arguments.getQueryLogStart() == null) {
+        throw new MetadataDumperUsageException(
+            "Missing option --query-log-start must be specified.");
+      }
+      if (arguments.getQueryLogEnd() == null) {
+        LOG.info(
+            "Missing option --query-log-end will be defaulted to: "
+                + arguments.getQueryLogEndOrDefault());
+      }
+
+      LOG.info(
+          "Log entries from {} to {} will be exported in increments of {}.",
+          arguments.getQueryLogStart(),
+          arguments.getQueryLogEndOrDefault(),
+          DurationFormatUtils.formatDurationWords(duration.toMillis(), true, true));
+      return createZonedIntervals(
+          arguments.getQueryLogStart(), arguments.getQueryLogEndOrDefault(), duration, truncator);
+    }
+
+    final int daysToExport = arguments.getQueryLogDays(7);
+    if (daysToExport <= 0)
+      throw new MetadataDumperUsageException(
+          "At least one day of query logs should be exported; you specified: " + daysToExport);
+
+    LOG.info(
+        "Log entries within the last {} days will be exported in increments of {} seconds.",
```

Comment:
The `seconds` word can be removed, because it is already generated by the `DurationFormatUtils.formatDurationWords` invocation.